preprocessing/text/cleaners.py

```diff
@@ -0,0 +1,128 @@
+import re
+
+from preprocessing.text.symbols import _alphabet, _punctuations, _not_end_punctuation, _numbers
+from preprocessing.text.numbers import Numbers
+
+
+class Cleaner:
```

Comment:
I think the idea of stratifying cleaning with simple objects is nice. But I find the inheritence here a little bit too engineered and not needed - I would give more freedom to the classes by just having a simple cleaning interface and no default implementation that is inherited. Generally composition over inheritance as it is much easier to maintain.

Reply:
Also it is not clear what needs to be overridden, it would be better to use and ABC for the top class and annotate the abstract method to make explicit.